File: tasks/task12_1/Point.kt
import kotlin.math.hypot

class Point(var x: Double, var y: Double) {
    fun distance() = hypot(x, y)
    fun distanceTo(point: Point) = hypot(x - point.x, y - point.y)
}

fun main() {
    val p = Point(4.5, 7.0)

    print("Enter x-coordinate: ")
    val x = readln().toDouble()

    print("Enter y-coordinate: ")
    val y = readln().toDouble()

    val uPoint = Point(x, y)

    println("Distance from origin ${uPoint.distance()}")
    println("Distance from (4.5, 7.0) ${"%.3f".format(uPoint.distanceTo(p))}")
}

File: tasks/task12_1/point.py
from math import hypot

class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y
    
    def distance(self):
        return hypot(self.x, self.y)
    
    def distanceTo(self, p):
        return hypot(self.x - p.x, self.y - p.y)

def main():
    p = Point(4.5, 7.0)

    x = float(input("Enter x-coordinate: "))
    y = float(input("Enter x-coordinate: "))
    
    uPoint = Point(x, y)

    print(f"Distance from origin {uPoint.distance()}")
    print(f"Distance from (4.5, 7.0) {round(uPoint.distanceTo(p), 3)}")

if __name__ == "__main__": main()